dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaManagerHandle.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2022-2024 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.connector.cloudera.manager;
+
+import com.google.common.collect.ImmutableList;
+import com.google.edwmigration.dumper.application.dumper.handle.Handle;
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+import javax.annotation.CheckForNull;
+import javax.annotation.concurrent.ThreadSafe;
+import org.apache.http.impl.client.CloseableHttpClient;
+
+@ThreadSafe
+public class ClouderaManagerHandle implements Handle {
+  private final URI apiURI;
+  private final CloseableHttpClient httpClient;
+
+  private List<ClouderaClusterDTO> clusters;
+
+  public ClouderaManagerHandle(URI apiURI, CloseableHttpClient httpClient) {
+    this.apiURI = apiURI;
+    this.httpClient = httpClient;
+  }
+
+  public URI getApiURI() {
+    return apiURI;
+  }
+
+  public URI getBaseURI() {
+    return apiURI.resolve("/");
+  }
+
+  public CloseableHttpClient getHttpClient() {
+    return httpClient;
+  }
+
+  @CheckForNull
+  public synchronized List<ClouderaClusterDTO> getClusters() {
+    return clusters;
```

Comment:
If `clusters` is null I suggest to either return an empty list or fail, whichever fits better.

Reply:
More importantly, this allows external code to do:
```
List<?> l1 = handle.getClusters();
l1.clear();
```
Which could be fixed by e.g. making a copy of `clusters`

Reply:
I agree, the field is updated to provide better readability for the code.

Reply:
Let's go back to the initial comment: I'd suggest to either fail or return an empty list if `clusters` wasn't initialized. This is a safer alternative to returning null.

Reply:
Could you explain what is the problem with returning `null`?  It's straightforward logic to validate and it's quite readable as well. 

Fail in this class is too much responsibilities for this class, the decision to fail should be done in particular task depends on the task logic. 
Empty list can be a good solution as null-object pattern, but in current case I need just simple check.

Reply:
Resolved. It works as expected if we will find more use cases we will change it.